docs/UserGuide.md

```diff
@@ -201,6 +202,45 @@ To check total modular credits:
 2. The result display box will display the message "Total MC has been calculated" along with your total modular credits amount.
 ![CheckMc2](images/CheckMc2.png)
 
+### Checking required modules: `required`
+
+The `required` command allows you to check the modules of the Computer Science curriculum you have yet to take, based on your current GradPad module list.
+
+Example: <br>
+You have added several modules into your GradPad so far but you are unsure if they are part of the Computer Science curriculum. You can simply check by using the `required` command.
+
+To check required modules:
+
+1. Type `required` into the command box, press **Enter** to execute it.
+
+2. The result display box will then display all the modules you have yet to take for each particular field, as shown below.
+
+### Checking Science modules: `science`
+
+The `science` command allows you to check all the available Science modules of the Computer Science curriculum.
+
+Example: <br>
+You wish to take a Science module to clear your Science requirement, but you forgot what modules are available. You can simply check by using the `science` command.
+
+To check Science modules:
+
+1. Type `science` into the command box, press **Enter** to execute it.
+
+2. The result display box will then display all the available Science modules that you can take, as shown below.
+
+### Searching for a module: `search`
+
+The `search` command allows you to search for any module available in NUS.
```

Comment:
Should we put a note or something letting users know that if their internet is down, then they can only search CS-curriculum modules?

Reply:
Oh ya good point, added in!